src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -11,35 +11,32 @@ namespace Microsoft.ML
     /// <summary>
     /// Specifies input and output column names for a transformation.
     /// </summary>
-    [BestFriend]
-    internal sealed class ColumnOptions
+    public sealed class InputOutputColumnPair
     {
-        private readonly string _outputColumnName;
-        private readonly string _inputColumnName;
+        /// <summary>
+        /// Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref="OutputColumnName"/> will be used as source.
+        /// </summary>
+        public readonly string InputColumnName;
+        /// <summary>
+        /// Name of the column resulting from the transformation of <see cref="InputColumnName"/>.
+        /// </summary>
+        public readonly string OutputColumnName;
 
         /// <summary>
         /// Specifies input and output column names for a transformation.
         /// </summary>
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
-        public ColumnOptions(string outputColumnName, string inputColumnName = null)
-        {
-            _outputColumnName = outputColumnName;
-            _inputColumnName = inputColumnName ?? outputColumnName;
-        }
-
-        /// <summary>
-        /// Instantiates a <see cref="ColumnOptions"/> from a tuple of input and output column names.
-        /// </summary>
-        public static implicit operator ColumnOptions((string outputColumnName, string inputColumnName) value)
+        public InputOutputColumnPair(string outputColumnName, string inputColumnName = null)
```

Comment:
>public InputOutputColumnPair(string outputColumnName, string inputColumnName = null) [](start = 8, length = 84)

Add another overload for the case when input name = output name. That would make it a lot clearer vs setting one to null.

Reply:
Unfortunately I believe that everywhere in the codebase we have the same pattern `string outputColumnName, string inputColumnName = null`. This is found in all the mlContext extensions for transforms. I can add it here, but I think it would make more sense to stick to the general pattern.

---
In reply to: [268726871](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268726871) [](ancestors = 268726871)

Reply:
We must be consistent. Sorry @glebuk!

---
In reply to: [268779384](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268779384) [](ancestors = 268779384,268726871)

Reply:
Ok  then.

---
In reply to: [268819893](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268819893) [](ancestors = 268819893,268779384,268726871)